shell/platform/darwin/ios/framework/Source/FlutterTextureRegistryRelay.h

```diff
@@ -0,0 +1,14 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/common/framework/Headers/FlutterMacros.h"
+#import "flutter/shell/platform/darwin/common/framework/Headers/FlutterTexture.h"
+
+#if FLUTTER_RUNTIME_MODE == FLUTTER_RUNTIME_MODE_DEBUG
+FLUTTER_DARWIN_EXPORT
+#endif
+@interface FlutterTextureRegistryRelay : NSObject <FlutterTextureRegistry>
```

Comment:
can you add a comment explaining that this relay is used to solve the issue with plugin retaining the engine?

Reply:
done.

Reply:
Use ```/**
 */```
comments as in the other headers.

From this comment I don't really understand when this would be used.  Returned from what?

Reply:
@jmagman, are these comments ok?

Reply:
How about:
```
/**
 * Wrapper around a weakly held collection of registered textures.
 *
 * Avoids a retain cycle between plugins and the engine.
 */
@interface FlutterTextureRegistryRelay : NSObject <FlutterTextureRegistry>

/**
 * A weak reference to a FlutterEngine that will be passed texture registration.
 */
@property(nonatomic, assign) NSObject<FlutterTextureRegistry>* parent;
```

Reply:
Thanks, this would be better.